src/reconstruction/fastGapFill/AuxillaryFiles/createXMatrix2.m

```diff
@@ -224,5 +224,6 @@
     end
     showprogress(i/length(compounds));
 end
-
-ExchangeRxnMatrix.mets = ExchangeRxnMatrix.mets';
+%Currently we are likely to have a largely blown up S matrix.
+ExchangeRxnMatrix.S = ExchangeRxnMatrix.S(1:numel(ExchangeRxnMatrix.mets),1:numel(ExchangeRxnMatrix.rxns));

```

Comment:
Is there actually a pre-allocation memory check done as well? Or is the code running until the memory is full?